ui/src/app/app-routing.module.ts

```diff
@@ -32,233 +31,242 @@ import {environment} from 'environments/environment';
 import {NavStore} from './utils/navigation';
 import {SignInComponent} from './views/sign-in/component';
 
+import * as fp from 'lodash/fp';
+
 declare let gtag: Function;
 
-const routes: Routes = [
+interface RouteExtension extends Route {
+  isPublic?: boolean;
+}
+
+declare type RoutesExtension = RouteExtension[];
```

Comment:
nit: this type does not seem necessary, I'd just use `RouteExtension[]` directly below